### PR TITLE
WEB-1531: Add Meta Data to Account Pages (SEO)

### PR DESCRIPTION
**Pull request by jeffkamo**
Add SEO helpful `<meta>` and `<title>` tags to the site header via the [`react-helmet`](https://www.npmjs.com/package/react-helmet) dependency.

**JIRA**: [WEB-1531](https://mobify.atlassian.net/browse/WEB-1531)

**Linked PRs**: #1009, and [progressive-web-sdk#866](https://github.com/mobify/progressive-web-sdk/pull/866)

## Changes
- New `pageMeta` key added to the following pages/containers:
    - Login page
    - Registration page
    - Account Dashboard
    - Account Information page
    - Account Address Book page
    - Account My Orders page
    - Account Order pages ("view order" for each individual order page)
    - Account Wishlist page

## Todos
- [x] Merge [progressive-web-sdk#866](https://github.com/mobify/progressive-web-sdk/pull/866)
- [ ] Merge #1009
- ~~(if applicable) analytics events instrumented to measure impact of change~~
- [ ] Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes
- Add `PageMeta` to more pages in future PRs
- [ ] Update SDK to latest version that includes the required SEO updates (see related PRs)

## How to test-drive this PR
**IMPORTANT:** run `npm i` to install new dependency
- Link the SDK to the scaffold
    - cd to the SDK repository
    - run `git checkout WEB-1531__more-pages-meta`
    - run `npm run prod:build`
    - run `npm link`
    - return to the scaffold by cd'ing to the `platform-scaffold` repository
    - `npm i` for `react-helmet` if you haven't done this yet
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Visit any of the affected pages (as listed above)
- **Verify** that the site `title` updates appropriately
- **Verify** that a `<meta name="description">` tag is present on that pages and matches what's on desktop
- **Verify** that a `<meta name="keywords">` tag is present on that pages and matches what's on desktop (in the case of Merlin's it will be empty)
- _Repeat above steps with the SFCC connector_

### CODE REVIEW

File: web/app/connectors/_merlins-connector/account/commands.js [hunk 1]
```diff
@@ -51,40 +58,60 @@ import {
...
         })
 }
 
 export const initAccountInfoPage = (url) => (dispatch) => {
     return dispatch(fetchPageData(url))
         .then((res) => {
             const [$, $response] = res
-            return dispatch(receiveAccountInfoData(parseAccountInfo($, $response)))
+            return Promise.all([
+                dispatch(receiveAccountInfoData(parseAccountInfo($, $response))),
```
Comment (jeremywiebe) [inline web/app/connectors/_merlins-connector/account/commands.js]: I'm pretty sure both of these actions are pure actions (not thunks) and as such we shouldn't need `Promise.all` on them... I think it would make sense not to return anything from inside here. We're already returning the `Promise` from fetchPageData so callers can properly wait on the completion of this command.

File: web/app/connectors/_merlins-connector/account/parsers.js [hunk 2]
```diff
@@ -106,9 +106,21 @@ const parseOption = ($option) => {
     }
 }
 
+export const parsePageMeta = ($, $response) => {
```
Comment (jeremywiebe) [inline web/app/connectors/_merlins-connector/account/parsers.js]: I suspect this parser is the same across all pages on Merlins... so would it make sense to move it up to a more general file? (maybe `/app/connectors/_merlins-connector/utils.js` or a new `/app/connectors/_merlins-connector/parsers.js`)?

File: web/app/connectors/_sfcc-connector/config.js [hunk 3]
```diff
@@ -41,3 +41,27 @@ export const getAccountInfoURL = () => `${getBaseURL()}Account-EditProfile`
...
+// These content IDs are useful for fetching content assets (like tiles,
+// descriptions, page meta, etc.) via the OCAPI "Content" resources. See the
+// following docs for more details on how to access the content:
+//
+//    Salesforce Commerce Cloud > Open Commerce API > Shop API resources > Content resource
+//
+// Content types can be found, created and edited in the SFCC backend via:
+//
+//     Merchant Tools > Content > Content Assets
+export const HOME_CONTENT_ID = 'home-main' // @TODO: replace with the real homepage content
```
Comment (jeremywiebe) [inline web/app/connectors/_sfcc-connector/config.js]: These will almost certainly vary across implementations. If you feel this PR is getting big we can make the change in a separate PR, but these really should be configurable from outside the connector (probably within the data passed into `registerConfig()` a few lines up)

File: web/app/connectors/_sfcc-connector/home/commands.js [hunk 4]
```diff
@@ -2,7 +2,15 @@
...
+
+import {makeApiRequest} from '../utils'
+import {HOME_CONTENT_ID} from '../config'
+
+import {parseHomeContent} from './parsers'
+
 export const initHomePage = () => (dispatch) => {
-    // Banners are being pulled from the bundle right now
-    return Promise.resolve()
+    return makeApiRequest(`/content/${HOME_CONTENT_ID}`, {method: 'GET'})
```
Comment (jeremywiebe) [inline web/app/connectors/_sfcc-connector/home/commands.js]: Let's DRY this up and make a utility function to fetch content by ID.

File: web/app/connectors/_sfcc-connector/config.js (hunk 3 above)
Comment (jeffkamo) [reply web/app/connectors/_sfcc-connector/config.js]: Indeed, will revisit this at a later date.

File: web/app/connectors/_merlins-connector/account/parsers.js (hunk 2 above)
Comment (jeffkamo) [reply web/app/connectors/_merlins-connector/account/parsers.js]: I put it in `parser-utils` for now, since that's where a ton of global parsers are at the moment. Although I agree to your last point, so this file should eventually be moved and renamed to `/app/connectors/_merlins-connector/parsers.js`

File: web/app/connectors/_sfcc-connector/account/commands.js [hunk 5]
```diff
@@ -35,15 +47,29 @@ import {
...
 import {addToCart} from 'progressive-web-sdk/dist/integration-manager/cart/commands'
 import {removeItemFromWishlist as removeItemFromWishlistCommand} from 'progressive-web-sdk/dist/integration-manager/account/commands'
 
 const initLoginData = () => (dispatch) => {
     dispatch(setSigninLoaded())
     dispatch(setRegisterLoaded())
-    return Promise.resolve()
+
+    return requestContentById(LOGIN_CONTENT_ID)
+        .then((res) => res.json())
```
Comment (jeremywiebe) [inline web/app/connectors/_sfcc-connector/account/commands.js]: Now that we have `requestContentById()` I think we could move the `.then((res) => res.json())` into it also.

Comment (jeffkamo) [reply web/app/connectors/_sfcc-connector/account/commands.js]: Will address this in #1037 instead of here :)
